### ISSUE TITLE
FIx CI to allow for easy managing of testing and master

### ISSUE THREAD

**Issue by mh4ckt3mh4ckt1c4s** [YunoHost-Apps/searxng_ynh#130]
Updates made to the CI in #113 by @ewilly to allow for multiple PR on both testing and master branches broke the CI and further fixes by me in #124 #127 made the CI run and create PR on testing but not successfully create the 2nd PR on master. Seems like the script used can not be used to create multiple PRs according to https://github.com/peter-evans/create-pull-request/issues/1302.

Reminder of the problem, trying to manage effectively automatic updates of the rolling release without having to manually merge from testing to master each time or merging directly into master and leaving testing out of sync.

Possible solution (that could even be implemented on all YNH app repos) could be to create another Action that would create a PR from testing to master when testing is ahead, preferably with a summary of new updates in the PR (including auto updates from upstream and potential merges from external contributors).

**Comment by ewilly** [comment]
Hi,
i just push #135 to: 

1. Restore the ci-auto-update PR to push them directly in master
2. Add a new workflow to create a PR to testing if behind master

can test that new workflow you will have to do it after the merge to master

**Comment by ewilly** [comment]
You can give me as admin access to assist yourself to maintain your app up-to-date

**Comment by mh4ckt3mh4ckt1c4s** [comment]
Hi, thanks for your PR. Not sure why you added an additionnal `testing-promotion` branch and not creating a PR directly from master to testing, but otherwise lgtm. 

Sorry but I think I don't have the rights to add you as a contributor to this repository. @ericgaspar can you do it ?

**Comment by mh4ckt3mh4ckt1c4s** [comment]
I just thought about it and I think it would be better the other way. Why sync testing from master when we can PR updates on testing as it was supposed to be ? We can create a PR to merge from testing to master once the "update PR" (or another improvement) is merged into testing ?

**Comment by ericgaspar** [comment]
@mh4ckt3mh4ckt1c4s I have updated your role as maintainer. you should have rights to add contributors.

**Comment by mh4ckt3mh4ckt1c4s** [comment]
@ericgaspar I now have access to the settings of the repo but not to the collaborators panel...

**Comment by ewilly** [comment]
Hi @mh4ckt3mh4ckt1c4s ,

just to be sure before commiting, here are the 2 proposed solutions. 

**SOLUTION A:** (the curent one)
1. Set the ci-auto-update PR to push them directly in master
2. Add a workflow to create a PR to testing if behind master

**SOLUTION B:** (the standard way in most of yunohost repositories)
1. Set the ci-auto-update PR to push them directly in testing
2. Add a workflow to create a PR to master if behind testing

Which one have your preference?

**Comment by mh4ckt3mh4ckt1c4s** [comment]
@ewilly you got it right. I'm for the solution B as it is consistent with the rest of YNH, and even improve existing workflow as non-updates PRs merged into testing will have their dedicated testing -> master PR created too.

**Comment by ewilly** [comment]
Done and merged !
